### PR TITLE
feat: create template no data content in dropdown #526

### PR DESCRIPTION
**Pull request by PedroMiguel7**
feat #526

noData template for when there are no options in the dropdown

Dropdown noData: 
image

In the Select: 
image

In the Chip with search: 
image

### CODE REVIEW

File: projects/ion/src/lib/no-data/no-data.module.ts
Comment (jAllanOli) [inline projects/ion/src/lib/no-data/no-data.module.ts:1]: is this file necessary since the declaration was moved to the shared module?

File: projects/ion/src/lib/dropdown/dropdown.component.spec.ts [hunk 1]
```diff
@@ -497,3 +497,55 @@ describe('IonDropdownComponent / Required', () => {
...
+    await sut(defaultNoData);
+    const element = screen.getByTestId('ion-no-data');
+    expect(element).toBeInTheDocument();
+    expect(screen.getByText('Não há dados')).toBeInTheDocument();
+    expect(
+      document.getElementById('ion-icon-exclamation-rounded')
+    ).toBeInTheDocument();
+  });
+
+  it('Should not render NoData when there are options.', async () => {
```
Comment (Deeved) [inline projects/ion/src/lib/dropdown/dropdown.component.spec.ts:520]: I believe this test is not necessary, as there are already tests that validate this

File: projects/ion/src/lib/dropdown/dropdown.component.spec.ts [hunk 2]
```diff
@@ -497,3 +497,55 @@ describe('IonDropdownComponent / Required', () => {
...
+    await sut({
+      ...defaultNoData,
+      options: [{ label: 'Option 1', selected: false }],
+    });
+    const element = screen.queryByTestId('ion-no-data');
+    expect(element).not.toBeInTheDocument();
+  });
+
+  it('Should render NoData with default parameters.', async () => {
```
Comment (Deeved) [inline projects/ion/src/lib/dropdown/dropdown.component.spec.ts:529]: This test is practically the same as the 510 line

File: projects/ion/src/lib/dropdown/dropdown.component.spec.ts (hunk 1 above)
Comment (PedroMiguel7) [reply projects/ion/src/lib/dropdown/dropdown.component.spec.ts:520]: ok, I'm going to delete the it that is repeating the validation of the default parameters

File: projects/ion/src/lib/dropdown/dropdown.component.spec.ts (hunk 2 above)
Comment (PedroMiguel7) [reply projects/ion/src/lib/dropdown/dropdown.component.spec.ts:529]: ok, I'm going to delete the it that is repeating the validation of the default parameters

File: projects/ion/src/lib/dropdown/dropdown.component.spec.ts (hunk 1 above)
Comment (PedroMiguel7) [reply projects/ion/src/lib/dropdown/dropdown.component.spec.ts:520]: I did the test to validate if noData was rendered with the default parameters because the storybook wasn't showing, I had to manually pass the noData configuration in the story.

File: projects/ion/src/lib/no-data/no-data.module.ts
Comment (PedroMiguel7) [reply projects/ion/src/lib/no-data/no-data.module.ts:1]: As per the design pattern, it is common for components to have their own module. For a more detailed explanation, I recommend referring to the following link:  #449

Comment (jAllanOli) [reply projects/ion/src/lib/no-data/no-data.module.ts:1]: Got it, thanks for the enlightenment.

**Review by iurynogueira — APPROVED** [review]

**Review by Deeved — APPROVED** [review]
